fvm/state/accounts.go

```diff
@@ -41,6 +42,34 @@ func NewAccounts(stateHolder *StateHolder) *Accounts {
 	}
 }
 
+func (a *Accounts) AllocateStorageIndex(address flow.Address) (uint64, error) {
+	var index uint64
+	indexBytes, err := a.getValue(address, false, KeyStorageIndex)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(indexBytes) == 0 {
+		// if not exist for the first time set it to zero and return
+		index = 0
```

Comment:
If I understand correctly, this function returns `StorageIndex` of `address` and both values are used to construct `StorageID`.

Currently storage index 0 is `StorageIndexUndefined`.  This is a special case because `StorageID` 0 is `StorageIDUndefined` (used to indicate absent of prev/next slab sibling), and `Address` 0 is `AddressUndefined` (used to indicate absent of owner).  

If we want to use storage index 0, owner can't be 0 at the same time because we would interpret the constructed storage id as undefined.

Reply:
thats a valid point, I'll update it.